drivers/gpio/gpio_pca953x.c

```diff
@@ -428,7 +428,7 @@ static int gpio_pca953x_init(const struct device *dev)
 
 		k_work_init(&drv_data->work, gpio_pca953x_work_handler);
 
-		rc = gpio_pin_configure_dt(&cfg->gpio_int, GPIO_INPUT);
+		rc = gpio_pin_configure_dt(&cfg->gpio_int, GPIO_INPUT | GPIO_ACTIVE_LOW);
```

Comment:
Proper polarity of the line should be defined in the flags cell in devicetree, for a given board, like here:
https://github.com/zephyrproject-rtos/zephyr/blob/e3ed78335945219e25a1e853173a1c1396afecda/boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts#L163
It's not a good idea to override it in the driver code.